src/classes/BGE_Wizard_CTRL.cls

```diff
@@ -0,0 +1,307 @@
+                model.requireTotalMatch = recordDetails.RequireTotalMatch__c;
+                model.activeFields = recordDetails.Active_Fields__c;
+
+            } else if (sObjectName == 'DataImportBatch__c') {
+                DataImportBatch__c recordDetails = [SELECT
+                        Name,
+                        Id,
+                        Batch_Description__c,
+                        EnableTotalEntry__c,
+                        RequireTotalMatch__c,
+                        Active_Fields__c
+                FROM DataImportBatch__c
+                WHERE Id = :recordId];
+
+                model.name = recordDetails.Name;
+                model.description = recordDetails.Batch_Description__c;
+                model.enableTotalEntry = recordDetails.EnableTotalEntry__c;
+                model.id = recordDetails.Id;
+                model.requireTotalMatch = recordDetails.RequireTotalMatch__c;
+                model.activeFields = recordDetails.Active_Fields__c;
+            }
+        }
+
+        model.templateFields = getAvailableFields();
+        model.labels = getLabels(sObjectName);
+
+        return JSON.serialize(model);
+    }
+
+    /*******************************************************************************************************
+    * @description Retrieves the related list of fields.
+    * @return list of Template Fields with sObjectName and Name.
+    */
+    public static List<AvailableField> getAvailableFields() {
+
+        List<AvailableField> fields = new List<AvailableField>();
+
+        Map<String, Schema.DescribeFieldResult> fieldMap = UTIL_Describe.getAllFieldsDescribe(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'));
+        Set<String> bannedFields = new Set<String>{'Account1Imported__c',
+                'Account1ImportStatus__c',
+                'Account2Imported__c',
+                'Account2ImportStatus__c',
+                'ApexJobId__c',
+                'Contact1Imported__c',
+                'Contact1ImportStatus__c',
+                'Contact2Imported__c',
+                'Contact2ImportStatus__c',
+                'DonationImported__c',
+                'DonationImportStatus__c',
+                'Donation_Possible_Matches__c',
+                'FailureInformation__c',
+                'HomeAddressImported__c',
+                'HomeAddressImportStatus__c',
+                'HouseholdAccountImported__c',
+                'ImportedDate__c',
+                'NPSP_Data_Import_Batch__c',
+                'PaymentImported__c',
+                'PaymentImportStatus__c',
+                'Payment_Possible_Matches__c',
+                'Status__c',
+                'Account2ImportStatus__c'
+        };
+
+        Set<String> allowedObjects = new Set<String>{
+                'Opportunity',
+                'Payment'
+        };
+
+        // these fields are required only if at least 1 field from its object group is selected
+        Set<String> systemRequiredFields = new Set<String>{
```

Comment:
We're not actually requiring account name/contact last name when we're using lookups. Is this still valid?

Reply:
I'd like to leave this logic in, since we do plan to need it when we come back to not using lookups any longer.